terror_error.go

```diff
@@ -241,13 +278,15 @@ type jsonError struct {
 // and the original global registry would be removed here.
 // This function is reserved for compatibility.
 func (e *Error) MarshalJSON() ([]byte, error) {
+	class, ec := strings.Split(string(e.codeText), ":"), 0
+	if len(class) > 0 {
```

Comment:
The `len(class) > 0` alway true, see: https://play.golang.org/p/QVD3mb336kr
So you can simplify as `ec := rfcCode2class[strings.Split(string(e.codeText), ":")[0]]`